src/Features/Core/Portable/CodeRefactorings/SyncNamespace/SyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.SyncNamespace), Shared]
+    internal sealed class SyncNamespaceCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var textSpan = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var service = document.GetLanguageService<ISyncNamespaceService>();
+            var actions = await service.GetRefactoringAsync(document, textSpan, cancellationToken).ConfigureAwait(false);
+
+            context.RegisterRefactorings(actions);
```

Comment:
i don't understaand the purpose of hte service.  why is this refactoring just calling out to it when it could just contain the code from teh service here...

Reply:
this comment still applies.  I understand why you have ISyncNamespaceService.TryGetReplacementReferenceSyntax, but i don't see why you have ISyncNamespaceService.GetRefactoringsAsync

Reply:
It was the result of a quick attempt of changing the refactoring into a language service, after I realized I need to handle references to affect types from a VB project. I will clean thigns up in terms of API as part of #30896, the shape of the service API will become clearer then.

Also there's a precedence, see `IMoveTypeService`. (not saying this is good, I will change it :))